### ISSUE TITLE
Callable class constructors

### ISSUE THREAD

**Issue by callumlocke** [babel/eslint-plugin-babel#34]
There's a stage 1 proposal for [callable class constructors](https://github.com/tc39/ecma262/blob/master/workingdocs/callconstructor.md) that allows code like this:

```js
import { initializeDate, ToDateString } from './date-implementation';

class Date {
  // `new Date()`
  constructor(...args) {
    initializeDate(super(), ...args);
  }

  // `Date()`
  call constructor() {
    return ToDateString(clockGetTime());
  }
}
```

And there's a [babel transform](http://babeljs.io/docs/plugins/transform-class-constructor-call/) for it.

But eslint doesn't seem to be able to handle it, even with this plugin. Am I configuring it wrong, or is this not supported?

**Comment by hzoo** [comment]
This plugin only adds rules. Are you using https://github.com/babel/babel-eslint?

**Comment by callumlocke** [comment]
Ah ok. Yes I've got `parser: babel-eslint` in my eslint config, and my babel config includes the transform-class-constructor-call plugin, and everything works fine with most Babel stuff, but then when I add `call constructor() {...}` to a class, eslint errors out with "Unexpected token". Should I report this over at babel-eslint then?

**Comment by hzoo** [comment]
Yeah technically not related to this but we can just do it here - most likely you aren't using the version that has babel 6 support (that is when it was added in). Right now its a beta version

**Comment by callumlocke** [comment]
No, I'm using the latest versions of everything. It doesn't seem to be able to handle `call constructor` in a class.

```
    "babel-cli": "^6.3.17",
    "babel-eslint": "^5.0.0-beta6",
    "babel-plugin-transform-export-extensions": "^6.3.13",
    "babel-preset-es2015-node5": "^1.1.1",
    "eslint": "^1.10.3",
    "eslint-plugin-babel": "^3.0.0",
```

My previous comment was incorrect,  the error is *not* "Unexpected token". It is actually `Duplicate name "constructor"`, when I have both `constructor() {...}` and `call constructor() {...}` in a class. 

(The "Unexpected token" I got before was actually because I had misspelled "constructor" :) So I was doing something like `call conTstructor() {...}` and this was giving "Unexpected token", I guess because `call` can only be used with a constructor function.)

Anyway, it looks like the problem rule is [no-dupe-class-members](http://eslint.org/docs/rules/no-dupe-class-members). Would it make sense for eslint-plugin-babel to add `babel/no-dupe-class-members`, which would recognise `constructor` and `call constructor` as different?
